src/Core/Form/IdentifiableObject/DataHandler/SupplierFormDataHandler.php

```diff
@@ -0,0 +1,150 @@
+
+use PrestaShop\PrestaShop\Core\CommandBus\CommandBusInterface;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\Command\AddSupplierCommand;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\Command\EditSupplierCommand;
+use PrestaShop\PrestaShop\Core\Domain\Supplier\ValueObject\SupplierId;
+use PrestaShop\PrestaShop\Core\Image\Uploader\ImageUploaderInterface;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+/**
+ * Handles submitted supplier form data
+ */
+final class SupplierFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $bus;
+    /**
+     * @var ImageUploaderInterface
+     */
+    private $imageUploader;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     * @param ImageUploaderInterface $imageUploader
+     */
+    public function __construct(
+        CommandBusInterface $commandBus,
+        ImageUploaderInterface $imageUploader
+    ) {
+        $this->bus = $commandBus;
+        $this->imageUploader = $imageUploader;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function create(array $data)
+    {
+        if (!isset($data['shop_association']) || !$data['shop_association']) {
+            $data['shop_association'] = [];
+        }
+
+        /** @var SupplierId $supplierId */
+        $supplierId = $this->bus->handle(new AddSupplierCommand(
+            $data['name'],
+            $data['address'],
+            $data['city'],
+            $data['id_country'],
+            $data['is_enabled'],
+            $data['description'],
+            $data['meta_title'],
+            $data['meta_description'],
+            $data['meta_keyword'],
+            $data['shop_association'],
+            $data['address2'],
+            $data['post_code'],
+            $data['id_state'],
+            $data['phone'],
+            $data['mobile_phone']
+        ));
+
+        /** @var UploadedFile $uploadedFlagImage */
+        $uploadedLogo = $data['logo'];
+
+        if ($uploadedLogo instanceof UploadedFile) {
+            $this->imageUploader->upload($supplierId->getValue(), $uploadedLogo);
+        }
+
+        return $supplierId->getValue();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function update($supplierId, array $data)
+    {
+        /** @var UploadedFile $uploadedFlagImage */
+        $uploadedLogo = $data['logo'];
+        $logo = null;
+
+        if ($uploadedLogo instanceof UploadedFile) {
+            $this->imageUploader->upload($supplierId, $uploadedLogo);
+        }
+
+        $command = new EditSupplierCommand($supplierId);
+        $this->fillCommandWithData($command, $data);
+
+        $this->bus->handle($command);
+    }
+
+    /**
+     * Fills command with provided data
+     *
+     * @param EditSupplierCommand $command
+     * @param array $data
+     */
+    private function fillCommandWithData(EditSupplierCommand $command, array $data)
+    {
+        $command->setName((string) $data['name']);
```

Comment:
Why do you cast this one and not the others ?

Reply:
🆙

Reply:
this one doesn't need casting indeed.